src/client/I2PService.h

```diff
@@ -199,9 +199,7 @@ class TCPIPAcceptor : public I2PService {
   virtual std::shared_ptr<I2PServiceHandler> CreateHandler(
       std::shared_ptr<boost::asio::ip::tcp::socket> socket) = 0;
 
-  virtual std::string GetName() {
-    return "generic TCP/IP accepting daemon";
-  }
+  std::string GetName() const { return "generic TCP/IP accepting daemon"; }
```

Comment:
Probably this code is not hot to consider the optimization,
so it may not matter.
Had it been hot,
temporary strings would be there regardless of the RVO.
RVO with copy elision avoids one more temporary,
but the main one remains regardless.

Consider the user code:
```cpp
std::string name = object->GetName();  // Bad. Temporary string. Only use if ``name`` to be modified later (unlikely).
const std::string& name = object->GetName();  // Bad. The same temporary to const ref binding.
```
The temporary string is unavoidable for this API.
Traditionally, the return value would be expected to be ``const std::string&``.
With ``const char*``, it may avoid string construction at all
because the compile time literal name seems to the common case.